saleor/graphql/product/mutations/products.py

```diff
@@ -833,6 +833,15 @@ def clean_input(cls, info, instance, data):
         # from the schema, only "basePrice" should be used here.
         price = data.get("base_price", data.get("price"))
         if price is not None:
+            if price < 0:
```

Comment:
Should 0 be the correct price in Saleor? 
@patrys @maarcingebala

Reply:
Definitely, it is common to set a product to free

Reply:
But in most popular case free products have restrictions on the number of items. Currently, we are not able to set it on the product variant. But we can create Voucher\Sale with the price set to 0 and quantity restriction.

Reply:
Free should be supported, it's not likely for someone to use it for a physical shippable item.

Reply:
We should be able to set 0 as a product or variant price.